source/extended-json.rst

```diff
@@ -0,0 +1,618 @@
+=============
+Extended JSON
+=============
+
+:Spec: 587
+:Spec-ticket: SPEC-587
+:Title: Extended JSON
+:Author: Luke Lovett
+:Spec Lead: David Golden
+:Advisory Group: Mike O'Brien, Christian Kvalheim, Hannes Magnusson
+:Status: Draft
+:Type: Standards
+:Last Modified: January 23, 2017
+:Version: 1.0.0
+
+.. contents::
+
+--------
+
+Abstract
+========
+
+MongoDB extended JSON is a string format for representing BSON documents. This
+specification defines the canonical format for representing each BSON type in
+the extended JSON format. Thus, a tool that implements extended JSON will be
+able to parse the output of any tool that emits canonical extended JSON.
+
+META
+====
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”,
+“SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be
+interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`_.
+
+Terms
+=====
+
+*Historical extended JSON* - A string format based on the JSON standard that
+describes a BSON document. The historical extended JSON format does not describe
+a specific, standardized format, and many tools, drivers, and libraries
+implement extended JSON in conflicting ways.
+
+*Extended JSON* - A string format based on the JSON standard that describes BSON
+documents. This specification gives a formal definition to this format.
+
+Specification
+=============
+
+Parsers
+-------
+
+An extended JSON parser (hereafter just "parser") is a tool that transforms an
+extended JSON string into another representation, such as BSON or a
+language-specific data structure. A parser MUST accept all strings in the
+canonical extended JSON format. A parser MAY also accept strings that adhere to
+other formats, such as historical extended JSON formats emitted by old versions
+of mongoexport. A parser SHOULD support at least 200 levels of nesting in an
+extended JSON document but MAY set other limits on strings it can accept as
```

Comment:
I think it's clear it's referring to the JSON spec section 9.  Please, let's assume reasonable implementations and not try to lawyer up to stop Byzantine interpretations.